rhp/v4/server.go

```diff
@@ -1107,7 +1107,7 @@ func (s *Server) handleRPCVerifySector(stream net.Conn) error {
 
 	sector, err := s.sectors.ReadSector(req.Root)
 	if err != nil {
-		return rhp4.NewRPCError(rhp4.ErrorCodeBadRequest, err.Error())
+		return rhp4.NewRPCError(rhp4.ErrorCodeHostError, err.Error())
 	}
```

Comment:
Couldn't this be `return err`? Either `ReadSector` already returns `ErrSectorNotFound` in which case the status code is `ErrorCodeHostError` or it returns some arbitrary error in which case the caller will return a `ErrHostInternalError`. 

Regardless I got one thing that might be good to double check. The integration test shouldn't really care about the error returned here. Both errors should result in the integrity check failing since we don't want to rely on specific errors returned by the host so that they can't swindle themselves out of integration checks.

Reply:
Hm ok so in `handleRPCReadSector` we check if it exists first. I think the most correct and consistent impl. would be to perform this check as well, return ` rhp4.ErrSectorNotFound` and indeed return the plain `err` in case the read fails.

I'll make sure to double check the integration tests in indexd. We do currently check for the error that includes the error code. Having the error code be part of the string is quite error prone if you check for `strings.Contains`...


```
if exists, err := s.sectors.HasSector(req.Root); err != nil {
    return fmt.Errorf("failed to check sector: %w", err)
} else if !exists {
    return rhp4.ErrSectorNotFound
}
```